runtime/j9vm/j7vmi.c

```diff
@@ -2257,25 +2257,40 @@ JVM_StartThread(JNIEnv* jniEnv, jobject newThread)
 {
 	J9VMThread* currentThread = (J9VMThread*)jniEnv;
 	J9JavaVM* javaVM = currentThread->javaVM;
-	UDATA priority, isDaemon, privateFlags;
+	UDATA priority = 0;
+	UDATA isDaemon = 0;
+	UDATA privateFlags = 0;
 	j9object_t newThreadObject;
 	UDATA result;
 
 	javaVM->internalVMFunctions->internalEnterVMFromJNI(currentThread);
 
 	newThreadObject = J9_JNI_UNWRAP_REFERENCE(newThread);
+#if JAVA_SPEC_VERSION >= 19
+	j9object_t threadHolder = J9VMJAVALANGTHREAD_HOLDER(currentThread, newThreadObject);
+#endif /* JAVA_SPEC_VERSION >= 19 */
 
 	if (0 != (javaVM->runtimeFlags & J9RuntimeFlagNoPriorities)) {
 		priority = J9THREAD_PRIORITY_NORMAL;
 	} else {
+#if JAVA_SPEC_VERSION >= 19
+		if (NULL != threadHolder) {
+			priority = J9VMJAVALANGTHREADFIELDHOLDER_PRIORITY(currentThread, threadHolder);
+		}
+#else /* JAVA_SPEC_VERSION >= 19 */
 		priority = J9VMJAVALANGTHREAD_PRIORITY(currentThread, newThreadObject);
+#endif /* JAVA_SPEC_VERSION >= 19 */
 	}
```

Comment:
This can be simplified:
```
	if (J9_ARE_NO_BITS_SET(javaVM->runtimeFlags, J9RuntimeFlagNoPriorities)) {
#if JAVA_SPEC_VERSION >= 19
		if (NULL != threadHolder) {
			priority = J9VMJAVALANGTHREADFIELDHOLDER_PRIORITY(currentThread, threadHolder);
		}
#else /* JAVA_SPEC_VERSION >= 19 */
		priority = J9VMJAVALANGTHREAD_PRIORITY(currentThread, newThreadObject);
#endif /* JAVA_SPEC_VERSION >= 19 */
	}
```
* the flag should be tested with `J9_ARE_NO_BITS_SET`
* `priority` is initialized to `J9THREAD_PRIORITY_NORMAL` so those branches are not needed here